### PR TITLE
LoKI: New samples UI

### PR DESCRIPTION
**Pull request by umithardal**
This PR implements a new UI for samples. No functionality yet.

Additionally;

- Renames `loki_script_model` to `loki_data_model`.

### CODE REVIEW

File: nicos_ess/loki/gui/loki_data_model.py [hunk 1]
```diff
@@ -31,7 +31,7 @@
 from nicos.guisupport.qt import QAbstractTableModel, QModelIndex, Qt
 
 
-class LokiScriptModel(QAbstractTableModel):
+class LokiDataModel(QAbstractTableModel):
```
Comment (ebadkamil) [inline nicos_ess/loki/gui/loki_data_model.py:34]: `LokiDataModel` sounds very generic. `LokiScriptModel` was specific to scripting table. But if it sounds reasonable to you, I have no issues.

File: nicos_ess/loki/gui/loki_samples.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        self.optionalComboBox.setCurrentIndex(-1)
+        self._init_table_panel()
+
+    def _init_table_panel(self):
+        headers = [
+            self.permanent_columns[name]
+            for name in self.columns_in_order
+        ]
+
+        self.model = LokiDataModel(headers)
```
Comment (ebadkamil) [inline nicos_ess/loki/gui/loki_samples.py:68]: Ok now I see your point with renaming. Forget about my previous comment.

**Comment by mattclarke** [comment]
Don't move stuff into a "legacy directory" - if it not used then just delete it please.
If we ever need it back we can get it from git

**Comment by umithardal** [comment]
> Don't move stuff into a "legacy directory" - if it not used then just delete it please.
> If we ever need it back we can get it from git

Now, there are still modules which depends on these files. What I can do is to refactor the function for example `measdialogs.py` requires from `sampleconf.py` but I cannot make sense of it. Most of these files (seems to me) copied over and may change along the way. In the mean time, I thought, keeping them alive in a legacy folder would make more sense. If it does not, still, I suggest I move them back where they are rather than deleting them. The latter requires more in depth care.

**Comment by umithardal** [comment]
I have restore old file locations and removed legacy directory. Cleaning up should be done in a separate PR.

**Review by kmurica — APPROVED** [review]
Looks good to me.

**Pull request merged**